### PR TITLE
hide empty help label in antd form items

### LINKED ISSUE #2779: with antd v4.18.3 or newer help labels on form items are always displayed even if they are empty
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description
Antd version 4.18.3 introduced a small change (https://github.com/ant-design/ant-design/pull/33644/files#diff-2bf6ed805cec0d6717765729f24ab52e77149471de0b5ad3a568c9ad1c6c297bR228) that only hides the help label between the form item input and the description field if the value of the help property is undefined. Before the change any falsy value would do. Unfortunately this looks pretty weird in conjunction with the description field as the description is shifted away from the input in a way that they seem disjoint. It even looks as if the description belonged to the form item below it. Just to give you an impression of how that looks like, below is a screenshots of the same form with the antd versions 4.18.2 and 4.18.3 respectively:

Screenshot from 2022-03-25 03-25-19

The fix should be fairly simple. The expression in https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/antd/src/templates/FieldTemplate/index.js#L68 needs to be changed from
```js
help={(!!rawHelp && help) || (!!rawErrors && renderFieldErrors())}
```
to
```js
help={(!!rawHelp && help) || (!!rawErrors && renderFieldErrors()) || undefined}
```
or better still to
```js
help={rawHelp ? help : rawErrors ? randerFieldErrors() : undefined}
```
which is a bit easier on the eyes.

Here is a very simple example that reproduces the problem:
https://stackblitz.com/edit/react-5zxqns-zozml8?file=index.js

Here is the exact same example but with a version of antd (4.18.2) that didn't yet cause that behaviour:
https://stackblitz.com/edit/react-5zxqns-gy4vis?file=index.js

### Steps to Reproduce
Set up rjsf with a version of antd that is higher than 4.18.3 and use form items with descriptions.

#### Expected behavior
The help label is only shown if there is actual help content. The description field should be right below the form input element without a wide gap between them.

#### Actual behavior
An empty help label is shown if there is no help or error information. The description field is shifted down with a notable gap between the form input element and the description field.

### Version
    "@rjsf/antd": "^4.0.1",
    "@rjsf/core": "^4.0.1",
    "antd": "4.19.1",

### PR DESCRIPTION
**Pull request by momesana**
### Reasons for making this change

`fixes #2779`

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Review by emmanueljohn1 — APPROVED** [review]

**Comment by momesana** [comment]
Is there anything else I could do to expedite the merge of this PR?

**Comment by epicfaace** [comment]
Thanks -- do you mind adding it to the weekly meeting agenda? I think a synchronous review would be easiest -- https://github.com/rjsf-team/react-jsonschema-form/issues/2677

File: packages/antd/src/templates/FieldTemplate/index.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ const FieldTemplate = ({
           colon={colon}
           extra={description}
           hasFeedback={schema.type !== 'array' && schema.type !== 'object'}
-          help={(!!rawHelp && help) || (!!rawErrors && renderFieldErrors())}
+          help={rawHelp ? help : rawErrors ? randerFieldErrors() : undefined}
```
Comment (epicfaace) [inline packages/antd/src/templates/FieldTemplate/index.js:68]: We also need to add a test here. Thanks!

**Comment by heath-freenome** [comment]
Closing as this was fixed indirectly via #3111 and #3112

**Pull request closed without merging**